### PR TITLE
image prune command fixed as per docker image prune.

### PR DESCRIPTION
**Pull request by kunalkushwaha**
makes `podman image prune` aligned with docker image prune command.

This PR makes following changes.
- default `podman image prune` now ask for confirmation from user with warning message. 
```
$ .podman image prune

WARNING! This will remove all dangling images.
Are you sure you want to continue? [y/N] 
```
- adds ``--force`` flag, which do not ask for any confirmation. Fixes https://github.com/containers/libpod/issues/4410

- adds `--filter` flag, which supports `label` & `until` format. Fixes https://github.com/containers/libpod/issues/4411
e.g.
```
$ podman image prune -a --filter until=2019-11-14T06:15:42.937792374Z             

WARNING! This will remove all dangling images.
Are you sure you want to continue? [y/N] 

$ podman image prune -a --filter until=10h                            

WARNING! This will remove all dangling images.
Are you sure you want to continue? [y/N] 
```

Signed-off-by: Kunal Kushwaha <kunal.kushwaha@gmail.com>

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
@kunalkushwaha you've a gofmt issue, you need to run:
`gofmt -s -w image/image.go` then add, commit and push.  Did you sign the commit too?
Anywho, look like a nice change overall, thanks for putting it together.

**Comment by baude** [comment]
you can also just run make validate

**Comment by kunalkushwaha** [comment]
@TomSweeneyRedHat @baude All tests are green now. PTAL

**Comment by rhatdan** [comment]
LGTM
@vrothberg @giuseppe @TomSweeneyRedHat @baude @QiWang19 @mheon @jwhonce PTAL

File: libpod/image/image.go [hunk 1]
```diff
@@ -74,6 +74,11 @@ type InfoImage struct {
 	Layers []LayerInfo
 }
 
+// Filter is a function to determine whether a image is included
+// in command output. Images to be outputted are tested using the function.
+// A true return will include the image, a false return will exclude it.
+type Filter func(*Image) bool
```
Comment (mheon) [inline libpod/image/image.go]: Can we rename this to `ImageFilter`? I can see us adding Container filters and Pod filters in the future, too, which might give us name collisions.

File: libpod/image/prune.go [hunk 2]
```diff
@@ -2,23 +2,92 @@ package image
...
+			return false
+		}, nil
+
+	}
+	return nil, nil
+}
+
 // GetPruneImages returns a slice of images that have no names/unused
-func (ir *Runtime) GetPruneImages(all bool) ([]*Image, error) {
+func (ir *Runtime) GetPruneImages(all bool, filters []string) ([]*Image, error) {
```
Comment (mheon) [inline libpod/image/prune.go]: I'd prefer to have this accept a `[]ImageFilter` instead, and converting from string to filter earlier.

**Comment by TomSweeneyRedHat** [comment]
code changes LGTM other than @mheon's comments.
Missing changes to man pages(docs/source/markdown/podman-image-prune.1.md) and bash completions (completions/bash/podman).

**Comment by TomSweeneyRedHat** [comment]
and most importantly, THANKS @kunalkushwaha for this pr.  It will be a nice add to Podman once complete.

File: libpod/image/image.go (hunk 1 above)
Comment (kunalkushwaha) [reply libpod/image/image.go]: I initially named this `Filter` as `ImageFilter`. But `make validate` reports error as its part of image package and user need to user `image.ImageFilter`.

Comment (rhatdan) [reply libpod/image/image.go]: Can we either tell validate to be quiet or make it plural
ImagesFilter

Comment (kunalkushwaha) [reply libpod/image/image.go]: Done. used `//nolint` for ignoring the check against `ImageFilter`

**Comment by TomSweeneyRedHat** [comment]
Our old friend the prune remote test is failing again, restarting.

**Comment by TomSweeneyRedHat** [comment]
LGTM if we can get the tests happy.  Current test issues I don't believe are related to this PR.

**Pull request merged**